src/tests/integration/MetaMorpho.t.sol

```diff
@@ -0,0 +1,300 @@
+            }
+        }
+    }
+
+    // --- Tests ---
+    function testDepositInOneMarket(uint256 d3mDeposit) public {
+        uint256 marketSupplyBefore = morpho.market(marketParams.id()).totalSupplyAssets;
+        uint256 spDaiTotalSupplyBefore = spDai.totalSupply();
+        uint256 spDaiTotalAssetsBefore = spDai.totalAssets();
+        uint256 spDaiMaxDepositBefore = spDai.maxDeposit(address(pool));
+        uint256 morphoBalanceBefore = dai.balanceOf(address(morpho));
+        uint256 daiTotalSupplyBefore = dai.totalSupply();
+
+        assertEq(spDai.totalSupply(), 0);
+        assertEq(spDai.totalAssets(), 0);
+        assertEq(plan.targetAssets(), 0);
+        assertEq(spDai.maxDeposit(address(pool)), type(uint184).max);
+
+        // Set target assets at `d3mDeposit` and exec.
+        vm.prank(operator);
+        plan.setTargetAssets(d3mDeposit);
+        hub.exec(ilk);
+
+        (,,, uint256 line,) = VatAbstract(vat).ilks(ilk);
+        uint256 depositedAssets = min(d3mDeposit, line * 1e18 / 1e45);
+        
+        assertEq(plan.targetAssets(), d3mDeposit);
+        
+        assertEq(morpho.market(marketParams.id()).totalSupplyAssets, marketSupplyBefore + depositedAssets);
+        
+        assertEq(spDai.balanceOf(address(pool)), depositedAssets);
+        assertEq(spDai.totalSupply(), spDaiTotalSupplyBefore + depositedAssets);
+        assertEq(spDai.totalAssets(), spDaiTotalAssetsBefore + morpho.expectedSupplyAssets(marketParams, address(spDai)));
+        assertEq(spDai.maxDeposit(address(pool)), spDaiMaxDepositBefore - depositedAssets);
+        
+        assertEq(dai.balanceOf(address(morpho)), morphoBalanceBefore + depositedAssets);
+        assertEq(dai.totalSupply(), daiTotalSupplyBefore + depositedAssets);
+    }
+
+    function testDepositInCappedMarket(uint184 cap, uint256 d3mDeposit) public {
+        uint256 marketSupplyBefore = morpho.market(marketParams.id()).totalSupplyAssets;
+
+        setCap(marketParams, cap);
+        
+        // Set target assets at `d3mDeposit` and exec.
+        vm.prank(operator);
+        plan.setTargetAssets(d3mDeposit);
+        hub.exec(ilk);
+
+        (,,, uint256 line,) = VatAbstract(vat).ilks(ilk);
+        uint256 depositedAssets = min(d3mDeposit, min(line * 1e18 / 1e45, cap));
+        
+        assertEq(morpho.market(marketParams.id()).totalSupplyAssets, marketSupplyBefore + depositedAssets);
+    }
+
+    function testWithdrawLiquid(uint256 target1, uint256 target2) public {
+        uint256 marketSupplyStart = morpho.market(marketParams.id()).totalSupplyAssets;
+        
+        (,,, uint256 line,) = VatAbstract(vat).ilks(ilk);
+        vm.assume(target1 < line * 1e18 / 1e45);
+        
+        adjustDebt(int256(target1));
+        
+        vm.assume(target2 <= pool.assetBalance());
+        
+        uint256 marketSupplyMiddle = morpho.market(marketParams.id()).totalSupplyAssets;
+        
+        uint256 depositedAssets1 = min(target1, line * 1e18 / 1e45);
```

Comment:
There is a `vm.assume` that `target1` is less than the `line`, so this will be target1.
I'd remove `depositedAssests1`, just use `target1` and rename `depositedAssets2` to `depositedAssets`.

Reply:
fixed